src/program/BlockStatement.js

```diff
@@ -176,6 +176,53 @@ export default class BlockStatement extends Node {
 
 		super.transpile(code, transforms);
 
+		if (transforms.asyncAwait && this.isFunctionBlock && this.parent.async && this.body.length) {
+			const first = this.body[0];
+			const last = this.body[this.body.length - 1];
+			const hasOnlyOneLine = this.body.length === 1;
+
+			// TODO refactor :)
```

Comment:
Refactor how? Who wrote this comment?

Better refactor now before merging, as later it will most likely not ever be refactored.